skillmap/src/components/UserProfile.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import { SkillMapState } from "../store/reducer";
+
+import { Modal } from './Modal';
+
+import { dispatchCloseUserProfile, dispatchShowDeleteAccountModal } from "../actions/dispatch"
+
+interface UserProfileProps {
+    signedIn: boolean;
+    profile: pxt.auth.UserProfile
+    showProfile: boolean;
+    dispatchCloseUserProfile: () => void;
+    dispatchShowDeleteAccountModal: () => void;
+}
+
+
+export class UserProfilImpl extends React.Component<UserProfileProps, {}> {
+    render() {
+        const { showProfile } = this.props;
+
+        if (showProfile) {
+            return this.renderUserProfile();
+        }
+
+        return <div/>
+    }
+
+    renderUserProfile = () => {
+        const user = this.props.profile;
+
+        return <Modal title={user?.idp?.displayName || ""} fullscreen={true} onClose={this.handleOnClose}>
+        <div className="profiledialog">
+            {this.getAccountPanel()}
+            {this.getFeedbackPanel()}
+        </div>
+        </Modal>
+    }
+
+    getAccountPanel = () => {
+        const { profile } = this.props;
+        const provider = profile?.idp?.provider && pxt.auth.identityProvider(profile?.idp?.provider);
+
+        const avatarElem = (
+            <div className="profile-pic avatar">
+                <img src={profile?.idp?.picture?.dataUrl} alt={lf("User")} />
+            </div>
+        );
+        const initialsElem = (
+            <div className="profile-pic avatar">
+                <span>{pxt.auth.userInitials(profile)}</span>
+            </div>
+        );
+
+        return <div className="account panel ui card">
+            <div className="header-text">
+                <label>{lf("Profile")}</label>
+            </div>
+            {profile?.idp?.picture?.dataUrl ? avatarElem : initialsElem}
+            <div className="row-span-two">
+                <label className="title">{lf("Name")}</label>
+                <p className="value">{profile?.idp?.displayName}</p>
+            </div>
+            <div className="row-span-two">
+                <label className="title">{lf("Username")}</label>
+                <p className="value">{profile?.idp?.username}</p>
+            </div>
+            <div className="row-span-two">
+                <label className="title">{lf("Provider")}</label>
+                <p className="value">{provider?.name}</p>
+            </div>
+            <div className="row-span-two" onClick={this.handleSignout}>
+                <div className="sign-out ui icon button" >
+                    <span className={`xicon ${profile?.idp?.provider}`} />
+                    <span> {lf("Sign out")} </span>
```

Comment:
I ran this locally and this sign out button looked like the buttons from the webapp. Felt inconsistent with the rest of skillmap's design. (we should bring them together)
![image](https://user-images.githubusercontent.com/12176807/129431645-734a918b-8c3d-4a81-8410-89b877846c46.png)

Reply:
yeah, but I thought since the header between skillmap/webapp are the same, user wouldn't really know the difference between skillmap and regular app? So it would be confusing if the profile looked different from before

Reply:
I could also change the regular app to match skillmap buttons?

Reply:
My 2c: I think it would be best for each to be consistent with its own style. Ideally they would share one common control lib and style sheet. But I don't know if that will ever happen due to the amount of reengineering.

Reply:
yeah most buttons in webapp already use the same component, we should at some point do the same in the skillmap--just add a button component and ensure that any buttons in the skillmap are using that component (shouldn't be TOO bad since the skillmap is small). at least this'll make it easier longer-term, if we eventually want to unify the two components or something

Reply:
maybe for this PR we could just make the user profile buttons look like the other skillmap buttons?